compiler/rustc_monomorphize/src/collector.rs

```diff
@@ -1143,8 +1155,16 @@ fn create_mono_items_for_vtable_methods<'tcx>(
                         Some(*instance).filter(|instance| should_codegen_locally(tcx, instance))
                     }
                 })
-                .map(|item| create_fn_mono_item(tcx, item, source));
-            output.extend(methods);
+                .map(|item| {
+                    let item = create_fn_mono_item(tcx, item, source);
+                    let inlined = if is_root {
```

Comment:
Where does this logic come from?

Reply:
Judging inlining does seem a little out of place here. I added the extend method to the MonoItems so it looks more reasonable